include/internal/ktls.h

```diff
@@ -196,11 +196,8 @@ static ossl_inline ossl_ssize_t ktls_sendfile(int s, int fd, off_t off,
     int ret;
 
     ret = sendfile(fd, s, off, size, NULL, &sbytes, flags);
-    if (ret == -1) {
-	    if (errno == EAGAIN && sbytes != 0)
-		    return sbytes;
-	    return -1;
-    }
+    if (ret == -1 && sbytes == 0)
```

Comment:
I think we need to initialize `sbytes = 0` if we're going to simplify the check like this -- kern_sendfile() will not copyout() to sbytes if the input offset is negative, so we are introducing a possibility for uninitialized memory usage.
But otherwise I agree that this is a good approach to take; it doesn't make sense for the kernel to return a nonzero sbytes if nothing was actually written, regardless of what error code is returned.

Reply:
Pushed a commit with this fix. Should I squash all subsequent modifications with the initial commit or should I keep them as is?

Reply:
Keep them as-is for now; they can be squashed during the merge process.
(BTW, the `git commit --fixup=<hash>` option can be useful for this sort of thing.)
Thanks!